File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int power = nums[0];
        for(int i = 1;i<nums.size();i++){
            power --;
            if (power < 0)return false;
            power = max(power, nums[i]);
        }
        return true;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        prefix = [1]
        for i in range(n):
            prefix.append(prefix[i] * nums[i])
        suffix = [1]
        for i in range(n):
            suffix.append(suffix[i] * nums[n - i - 1])
        ans = []
        for i in range(n):
            ans.append(prefix[i] * suffix[n - i - 1])
        return ans